src/main/java/com/hierynomus/smbj/Config.java

```diff
@@ -15,36 +15,199 @@
  */
 package com.hierynomus.smbj;
 
-import java.util.EnumSet;
-import java.util.List;
-import java.util.Random;
-import java.util.UUID;
 import com.hierynomus.mssmb2.SMB2Dialect;
+import com.hierynomus.mssmb2.SMB2Packet;
 import com.hierynomus.protocol.commons.Factory;
 import com.hierynomus.security.SecurityProvider;
+import com.hierynomus.security.jce.JceSecurityProvider;
 import com.hierynomus.smbj.auth.Authenticator;
+import com.hierynomus.smbj.auth.NtlmAuthenticator;
+import com.hierynomus.smbj.auth.SpnegoAuthenticator;
 
-public interface Config {
+import java.security.SecureRandom;
+import java.util.*;
 
-    Random getRandomProvider();
+public final class Config {
+    private Set<SMB2Dialect> dialects;
+    private List<Factory.Named<Authenticator>> authenticators;
+    private Random random;
+    private UUID clientGuid;
+    private boolean signingRequired;
+    private SecurityProvider securityProvider;
+    private int readBufferSize;
+    private int writeBufferSize;
+    private int transactBufferSize;
 
-    SecurityProvider getSecurityProvider();
+    public static Config createDefaultConfig() {
+        return builder()
+            .withDialect(SMB2Dialect.SMB_2_1)
```

Comment:
Add a `withDialects(List<>)` and a `withDialects(Dialect...)`, same for `withAuthenticators`. It makes the Config API more fluent, and also clearer that you can add multiple. Those methods should clear the existing set and replace it with the configured set.